mlflow/sagemaker/__init__.py

```diff
@@ -388,6 +394,17 @@ def deploy(
         **assume_role_credentials,
     )
 
+    if variant_name:
+        pattern = r"[a-zA-Z0-9](-*[a-zA-Z0-9]){0,62}"
+        if not re.fullmatch(pattern, variant_name):
+            raise MlflowException(
+            message=(
+                "Variant name must match the following pattern" 
+                " {variant_pattern}".format(variant_pattern=pattern)
+            ),
+            error_code=INVALID_PARAMETER_VALUE,
+        )
```

Comment:
Do we need to validate the variant name in the client, or can SageMaker perform the validation?

Reply:
We don't have to.  Sagemaker will throw an error out if it is invalid.  I can remove this if that is preferred

Reply:
Awesome! Yeah, I think we can remove this :)